internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,168 @@ package github_connection
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
 	"github.com/google/go-github/v61/github"
 )
 
-type GithubConnectionParams struct {
-	RepositoryName    string
-	Token             string
-	RepoOwner         string
-	PullRequestNumber int
+type GithubConnection struct {
+	Client            *github.Client `json:"client"`
+	RepositoryName    string         `json:"repository_name"`
+	RepoOwner         string         `json:"repo_owner,omitempty"`
+	PullRequestNumber int            `json:"pull_request_number"`
+	EventName         string         `json:"event_name"`
+}
+
+type FileChange struct {
+	Filename    string
+	LineNumbers []int
+	Content     string
 }
 
-func NewGithubConnection(params GithubConnectionParams) (*GithubConnection, error) {
+func NewGithubConnection() (*GithubConnection, error) {
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("GITHUB_TOKEN environment variable is not set")
+	}
+
+	repoFullName := os.Getenv("GITHUB_REPOSITORY")
+	if repoFullName == "" {
+		return nil, fmt.Errorf("GITHUB_REPOSITORY environment variable is not set")
+	}
+	repoParts := strings.Split(repoFullName, "/")
+	if len(repoParts) != 2 {
+		return nil, fmt.Errorf("invalid GITHUB_REPOSITORY format: %s", repoFullName)
+	}
+	repoOwner := repoParts[0]
+	githubRepositoryName := repoParts[1]
+
+	pullRequestNumberStr := os.Getenv("GITHUB_PR_NUMBER")
+	if pullRequestNumberStr == "" {
+		return nil, fmt.Errorf("GITHUB_PR_NUMBER environment variable is not set")
+	}
+	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
+	}
 
```

Comment:
Code Review 2024-08-18 01:18:35 - New lines 18 to 42:

func NewGithubConnection() (*GithubConnection, error) {
	token := os.Getenv("GITHUB_TOKEN")
	if token == "" {
		return nil, fmt.Errorf("GITHUB_TOKEN environment variable is not set")
	}

	repoFullName := os.Getenv("GITHUB_REPOSITORY")
	if repoFullName == "" {
		return nil, fmt.Errorf("GITHUB_REPOSITORY environment variable is not set")
	}
	repoParts := strings.Split(repoFullName, "/")
	if len(repoParts) != 2 {
		return nil, fmt.Errorf("invalid GITHUB_REPOSITORY format: %s", repoFullName)
	}
	repoOwner := repoParts[0]
	githubRepositoryName := repoParts[1]

	pullRequestNumberStr := os.Getenv("GITHUB_PR_NUMBER")
	if pullRequestNumberStr == "" {
		return nil, fmt.Errorf("GITHUB_PR_NUMBER environment variable is not set")
	}
	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
	if err != nil {
		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
	}